src/main/java/frc/team2412/robot/subsystem/ShooterSubsystem.java

```diff
@@ -1,11 +1,135 @@
+
+    /**
+     * Constructor for shooter subsystem.
+     * 
+     * @param flywheelMotor1
+     *            The first motor connected to the flywheel
+     * 
+     * @param flywheelMotor2
+     *            The second motor connected to the flywheel
+     * 
+     * @param turretMotor
+     *            The motor that controls the horizontal rotation of the
+     *            turret
+     * 
+     * @param hoodMotor
+     *            The motor that controls the angle of the hood above the
+     *            turret
+     * 
+     */
+    public ShooterSubsystem(WPI_TalonFX flywheelMotor1, WPI_TalonFX flywheelMotor2, WPI_TalonFX turretMotor,
+            WPI_TalonFX hoodMotor) {
+        // Motor configs
+        var flywheelLimit = new SupplyCurrentLimitConfiguration(true, 40, 40, 500);
+        flywheelMotor1.configSupplyCurrentLimit(flywheelLimit);
+        flywheelMotor1.setNeutralMode(NeutralMode.Coast);
+        flywheelMotor2.configSupplyCurrentLimit(flywheelLimit);
+        flywheelMotor2.setNeutralMode(NeutralMode.Coast);
+        flywheelMotor2.setInverted(true);
+
+        var limit = new SupplyCurrentLimitConfiguration(true, 10, 10, 500);
+        TalonFXConfiguration turretConfig = new TalonFXConfiguration();
+        turretConfig.forwardSoftLimitThreshold = MAX_TURRET_ANGLE * DEGREES_TO_ENCODER_TICKS;
+        turretConfig.reverseSoftLimitThreshold = MIN_TURRET_ANGLE * DEGREES_TO_ENCODER_TICKS;
+        turretConfig.forwardSoftLimitEnable = true;
+        turretConfig.reverseSoftLimitEnable = true;
+        turretMotor.configSupplyCurrentLimit(limit);
+        turretMotor.configAllSettings(turretConfig);
+        turretMotor.setNeutralMode(NeutralMode.Brake);
+        turretMotor.configSelectedFeedbackSensor(TalonFXFeedbackDevice.IntegratedSensor, 0, 0);
+        turretMotor.config_kP(0, TURRET_P);
+        turretMotor.config_kI(0, TURRET_I);
+        turretMotor.config_kD(0, TURRET_D);
+        TalonFXConfiguration hoodConfig = new TalonFXConfiguration();
+        hoodConfig.forwardSoftLimitThreshold = MAX_HOOD_ANGLE * DEGREES_TO_ENCODER_TICKS;
+        hoodConfig.reverseSoftLimitThreshold = MIN_HOOD_ANGLE * DEGREES_TO_ENCODER_TICKS;
+        hoodConfig.forwardSoftLimitEnable = true;
+        hoodConfig.reverseSoftLimitEnable = true;
+        hoodMotor.configSupplyCurrentLimit(limit);
+        hoodMotor.configAllSettings(hoodConfig);
+        hoodMotor.setNeutralMode(NeutralMode.Brake);
+
+        this.flywheelMotor1 = flywheelMotor1;
+        this.flywheelMotor2 = flywheelMotor2;
+        this.turretMotor = turretMotor;
+        this.hoodMotor = hoodMotor;
+    }
+
+    public void hoodMotorSetAngle(double degrees) {
+        hoodMotor.set(ControlMode.Position, DEGREES_TO_ENCODER_TICKS * degrees);
+    }
+
+    // TODO make hardstop
+    public void hoodMotorStop() {
+        hoodMotor.set(STOP_MOTOR);
+    }
+
+    // TODO make sure these motors are going in the right direction
```

Comment:
I think this is "todone" from the motor configs above, so the comment can be deleted.